test/images/regression-issue-74839/tcp.go

```diff
@@ -133,16 +142,29 @@ func (t *tcpPacket) encode(src, dest net.IP, data []byte) []byte {
 
 func checksumTCP(src, dest net.IP, tcpHeader, data []byte) uint16 {
 	log.Printf("calling checksumTCP: %v %v %v %v", src, dest, tcpHeader, data)
-	chk := &tcpChecksumer{}
+	chk := &tcpChecksummer{}
 
 	// Encode pseudoheader.
-	chk.add(src.To4())
-	chk.add(dest.To4())
+	// IPv4 [ src (4) | dst (4) | rsv (1) | proto (1) | tcp length (2)] ... | tcp header | data
+	if src.To4() != nil {
+		chk.add(src.To4())
+		chk.add(dest.To4())
+		pseudoHeader := make([]byte, 4)
+		pseudoHeader[1] = tcpProtoNum
```

Comment:
you could just `chk.add([]byte{0, tcpProtoNum})` and then do a second add with just the length.
(I guess this is just moving the existing code unchanged, but it's not really clear why it adds the src/dest separately but then agglomerates the remaining fields together. Likewise in the IPv6 case.)

Reply:
I copied pasted as you said, I tested the result and it works, if we'll need to reuse it sometime later I would take a completely approach or us a more prove library like gopacket

Reply:
actually the pseudo header is including the source ips , as I show in the comment `[ src (4) | dst (4) | rsv (1) | proto (1) | tcp length (2) ]`

Reply:
The pseudo header includes the IPs. `pseudoHeader` doesn't. That's the thing that's weird.

Reply:
will change it too thanks

Reply:
ok, I think that this is this way to use the `binary.BigEndian.PutUint16()` method

Reply:
yeah, that should be fixed

Reply:
it seems I can not assign directly a `type IP`to a slice `[]bytes` :/

Reply:
will keep it as is if you don't mind,